### DISCUSSION TITLE
Swap size on a linux box?

### DISCUSSION THREAD

**Discussion by cadilhac** [ravendb/ravendb#14985] [Q&A] [answered]
Hi,

After updating a ravendb installation to 5.3.104 on a linux box, I got this in the Studio notifications center:

> The current swap size is '0 Bytes' and it is lower then the threshold defined '1 GBytes'

I never cared about swap and never noticed something bad in production.
How important is this? Should I set up some swap space on my linux box (which is a 2 GB RAM, 1 vCPU, 60 GB SSD AWS Lightsail machine with only ravendb in it)? If yes, how to do this without breaking anything?

Thanks

**Comment by ayende** [comment]
There are some scenarios where not having a swap is bad, for example: https://github.com/dotnet/runtime/issues/851

Looking at the root issue, it has been resolved.  https://github.com/dotnet/runtime/issues/851
We may want to remove that warning entirely. 

That said, there are still some advantages to having swap to have something to handle spike (but the downside is that a machine that swaps usually does nothing else).

@ppekrol @gregolsky - thoughts?

**Comment by gregolsky** [comment, marked as answer]
SWAP is not required and that hint can be disabled in the configuration by setting it to `PerformanceHints.Memory.MinSwapSizeInMb` to 0. 

For performance reasons on machines running databases it's recommended to set `vm.swappiness` kernel parameter to 0, so that it uses swap as a last resort - by default e.g. in Ubuntu it starts to use it when 60% of mem is in use.

However it's sometimes helpful to have some SWAP space in situations where you e.g. want to debug a memory leak - you will notice SWAP usage and that tells you then your memory is full - or as an indicator that your machine probably needs a RAM upgrade. As Oren mentioned it can be used as means to handle memory spikes. 

In general when OS starts using SWAP it usually slows things down. Note that having large amounts of SWAP in use on a weak cloud disk  may render the instance not responsive since swapping takes a lot of IO (assuming it's put on the same disk). It's usually more safe to set up a separate volume for SWAP alone, so that it does not consume database server's IOPS.
